### ISSUE TITLE
MarkerView, PointAnnotation do not reflect updates on map when coordinates change

### ISSUE THREAD

**Issue by sweatherall** [rnmapbox/maps#2200]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

iOS

### `@rnmapbox/maps` version

10.0.0-beta.33

### Standalone component to reproduce

```javascript
import React from 'react';
import { View, Text } from 'react-native';
import MapboxGL, {
  MapView,
  Camera,
} from '@rnmapbox/maps';

const coordinates = [
  [-73.99155, 40.73],
  [-73.99155, 40.74],
]

class BugReportExample extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      show: true,
    };
  }

  onPress = () => {
    this.setState({ show: !this.state.show });
  }

  render() {
    return (
      <MapView style={{ flex: 1 }} onPress={this.onPress}>
        <Camera centerCoordinate={coordinates[0]} zoomLevel={12} />

        <MapboxGL.ShapeSource
            id="currentLocationSource"
            shape={
              {
                type: 'Point',
                coordinates: coordinates[this.state.show ? 0 : 1],
              }
            }
          >
            <MapboxGL.CircleLayer
              id="currentLocationCircle"
              style={{
                circleOpacity: 0.5,
                circleColor: 'yellow',
                circleRadius: 30,
              }}
            />
          </MapboxGL.ShapeSource>

          <MapboxGL.PointAnnotation
            coordinate={coordinates[this.state.show ? 0 : 1]}
            id="pt-ann"
          >
            <View style={[styles.circle, {backgroundColor: 'red'}]}>
              <Text style={styles.text}>PA</Text>
            </View>
          </MapboxGL.PointAnnotation>

          <MapboxGL.MarkerView coordinate={coordinates[this.state.show ? 1 : 0]}>
            <View style={styles.circle}>
              <Text style={styles.text}>MV</Text>
            </View>
          </MapboxGL.MarkerView>
      </MapView>
    );
  }
}

const styles = {
  circle: {
    backgroundColor: 'blue',
    width: 40,
    height: 40,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: 'white',
    fontWeight: 'bold',
  },
};

export default BugReportExample;

```

### Observed behavior and steps to reproduce

This example demonstrates the difference in behavior between `ShapeSource` and `MarkerView` (and `PointAnnotation`) when the relevant position prop (`shape` or `coordinate`) gets updated.

All components are referencing coordinates depending on a state value. When the state changes, `ShapeSource` is the only component to accurately represent its new position on the map. `MarkerView` and `PointAnnotation` position stay static.

To reproduce:
* Have a simple state with a boolean value that can get toggled via an `onPress` action.
* For each component: dynamically reference a coordinate from an array of coordinates based on this boolean value.
* Click anywhere on the map to trigger the `onPress` action that will update the state.

In this example video, Yellow = `ShapeSource`, Red = `PointAnnotation`, Blue = `MarkerView`.
The `ShapeSource` is the only one that updates position, although they are all referencing a value based on a changing state value.

https://user-images.githubusercontent.com/4127588/189210419-6fe4205d-a205-4d0e-954e-d138e7ab05a6.mov

### Expected behavior

I would expect `MarkerView` and `PointAnnotation` to update their position on the map when the coordinate prop gets updated, in the same way that `ShapeSource` does when the shape prop gets updated. Is this possible?

### Notes / preliminary analysis

This is also demonstrated in `example/src/examples/V10/QueryTerrainElevation.js` - where the altitude `AnimatedMarkerView` display is not getting visually updated on the map, but the `Animated.ShapeSource` representing the current location is (they use the same `actPoint` value for their coordinate) .

Also FWIW, when I add some logs into `ios/RCTMGL-v10/RCTMGLMarkerView.swift`, I can see that it calls `_updateCoordinate` with the correct coordinates - but it visually does not update on the map.

### Additional links and references

Other potentially related issues: #2196 , #2115

**Comment by KiwiKilian** [comment]
For me it's now fixed in `10.0.0-beta.35` for `MarkerView`. I have not tested anything else.

**Comment by sweatherall** [comment]
@KiwiKilian thank you for the update. I just tested this issue with `v10.0.0-beta.38` and `MarkerView` seems to be updating its position as expected! 
`PointAnnotation` however still seems to have issues updating.
